presto-main-base/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -704,7 +713,23 @@ public Void visitCall(CallExpression call, Context context)
                 }
                 if (!isPushDownSubfieldsFromLambdasEnabled) {
                     context.setLambdaSubfields(Context.ALL_SUBFIELDS_OF_ARRAY_ELEMENT_OR_MAP_VALUE);
-                    call.getArguments().forEach(argument -> argument.accept(this, context));
+                    // map_subset(feature, constant_array) is only accessing fields specified in feature map.
+                    // For example map_subset(feature, array[1, 2]) is equivalent to calling element_at(feature, 1) and element_at(feature, 2) for subfield extraction
+                    if (isPushdownSubfieldsForMapSubsetEnabled && functionResolution.isMapSubSetFunction(call.getFunctionHandle()) && call.getArguments().get(0) instanceof VariableReferenceExpression && call.getArguments().get(1) instanceof ConstantExpression) {
```

Comment:
I am wondering if PushdownSubfields is the right place for this optimization because here only the context is changed and the name of this visitor also is **SubfieldExtractor**